lms/static/scripts/frontend_apps/components/InitialLoadingTracker.tsx

```diff
@@ -0,0 +1,71 @@
+import { Spinner, SpinnerOverlay } from '@hypothesis/frontend-shared';
+import classnames from 'classnames';
+import type { ComponentChildren } from 'preact';
+import { useCallback, useMemo, useState } from 'preact/hooks';
+
+import { InitialLoadingContext } from '../utils/initial-loading-context';
+
+export type InitialLoadingProps = {
+  children: ComponentChildren;
+  classes?: string | string[];
+};
+
+export default function InitialLoadingTracker({
+  children,
+  classes,
+}: InitialLoadingProps) {
+  const [loadingInstances, setLoadingInstances] = useState<
+    Record<string, boolean>
+  >({});
+  const initialLoadingInProgress = useMemo(() => {
+    // We consider loading is in progress as long as there are no loading
+    // instances or some of them are still in progress
+    const values = Object.values(loadingInstances);
+    return values.length === 0 || values.some(Boolean);
+  }, [loadingInstances]);
+  const updateLoadingForId = useCallback(
+    ({ id, loading }: { id: string; loading: boolean }) => {
+      // We ignore any loading change once the initial loading finished
+      if (!initialLoadingInProgress) {
+        return;
+      }
+
+      setLoadingInstances(prev => ({ ...prev, [id]: loading }));
+    },
+    [initialLoadingInProgress],
+  );
+  const startLoading = useCallback(
+    (id: string) => updateLoadingForId({ id, loading: true }),
+    [updateLoadingForId],
+  );
+  const finishLoading = useCallback(
+    (id: string) => updateLoadingForId({ id, loading: false }),
+    [updateLoadingForId],
+  );
+  const contextValue = useMemo(
+    () => ({ startLoading, finishLoading }),
+    [finishLoading, startLoading],
+  );
+
+  return (
+    <InitialLoadingContext.Provider value={contextValue}>
+      {initialLoadingInProgress && (
+        <div
+          className={classnames(
+            'flex items-center justify-center bg-white/50',
+            classes,
+          )}
+        >
+          <Spinner size="lg" />
+        </div>
```

Comment:
Not using a SpinnerOverlay, so that it doesn't overlap with the dashboard's header and footer.